### PR TITLE
Client - GeoNode Datasets Catalog Plugin

### PR DESCRIPTION
**Pull request by allyoucanmap**
This PR introduces following changes:

- dataset catalog plugin that uses the V2 API to map GeoNode resource to layer
- improve map workflow based on latest changes on V2 API

### CODE REVIEW

File: geonode_mapstore_client/client/js/api/geonode/v2/index.js [hunk 1]
```diff
@@ -214,6 +214,41 @@ export const getMaps = ({
...
+    pageSize = 20,
+    page = 1,
+    sort,
+    ...params
+}) => {
+    return requestOptions(DATASETS, () => axios
+        .get(
+            parseDevHostname(
+                addQueryString(endpoints[DATASETS], q && {
+                    search: q,
```
Comment (DavidQuartz) [inline geonode_mapstore_client/client/js/api/geonode/v2/index.js:228]:
```suggestion
                addQueryString(endpoints[DATASETS], query && {
                    search: query,
```

**Review by DavidQuartz — COMMENTED** [review]
Can we have a more descriptive name for `q` ?

**Review by DavidQuartz — COMMENTED** [review]
Can we have a more descriptive name for `q` ?

**Comment by allyoucanmap** [comment]
@DavidQuartz  thanks for the review, I'll work on next task and update the name of variable also on other component that have the same behavior and workflow. Going to merge this to make the maps work on the master instance

**Pull request merged**